File: src/main/java/sawyern/cookiebot/repository/AccountRepository.java
package sawyern.cookiebot.repository;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import sawyern.cookiebot.models.entity.Account;

import java.util.List;
import java.util.Optional;

@Repository
public interface AccountRepository extends CrudRepository<Account, String> {
    Optional<Account> findByDiscordId(String discordId);
    List<Account> findAll();
}

File: src/main/kotlin/sawyern/cookiebot/repository/AccountRepository.kt
package sawyern.cookiebot.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import sawyern.cookiebot.models.Account

@Repository
interface AccountRepository: JpaRepository<Account, String> {
    fun findByDiscordId(discordId: String): Account?
}